go/felix/set/set.go

```diff
@@ -88,3 +89,15 @@ func (set mapSet) Copy() Set {
 	}
 	return cpy
 }
+
+func (set mapSet) Equals(other Set) bool {
```

Comment:
Could you take advantage of `reflect.DeepEquals()` here?  I.e. `return reflect.DeepEquals(set, other)`?